pkg/templates/kubeadm/v1beta4/kubeadm.go

```diff
@@ -0,0 +1,621 @@
+	ClusterConfiguration *kubeadmv1beta4.ClusterConfiguration
+
+	KubeletConfiguration   runtime.Object
+	KubeProxyConfiguration runtime.Object
+}
+
+// NewConfig returns all required configs to init a cluster via a set of v1beta4 configs
+func NewConfig(s *state.State, host kubeoneapi.HostConfig) (*Config, error) {
+	cluster := s.Cluster
+
+	overwriteRegistry := ""
+	if cluster.RegistryConfiguration != nil {
+		overwriteRegistry = cluster.RegistryConfiguration.OverwriteRegistry
+	}
+
+	bootstrapToken, err := bootstraptokenv1.NewBootstrapTokenString(s.JoinToken)
+	if err != nil {
+		return nil, fail.Runtime(err, "generating kubeadm bootstrap token")
+	}
+
+	var advertiseAddress string
+	if cluster.ClusterNetwork.IPFamily.IsIPv6Primary() {
+		advertiseAddress = host.IPv6Addresses[0]
+	} else {
+		advertiseAddress = newNodeIP(host)
+	}
+
+	controlPlaneEndpoint := fmt.Sprintf("%s:%d", cluster.APIEndpoint.Host, cluster.APIEndpoint.Port)
+	certSANS := certificate.GetCertificateSANs(cluster.APIEndpoint.Host, cluster.APIEndpoint.AlternativeNames)
+
+	initConfig := newInitConfiguration(bootstrapToken, advertiseAddress)
+	joinConfig := newJoinConfiguration(advertiseAddress, s.JoinToken, controlPlaneEndpoint)
+	nodeRegistration := newNodeRegistration(s, host)
+	nodeRegistration.IgnorePreflightErrors = []string{
+		"DirAvailable--var-lib-etcd",
+		"DirAvailable--etc-kubernetes-manifests",
+		"ImagePull",
+	}
+
+	clusterConfig := &kubeadmv1beta4.ClusterConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "kubeadm.k8s.io/v1beta4",
+			Kind:       "ClusterConfiguration",
+		},
+		ClusterName:          cluster.Name,
+		KubernetesVersion:    cluster.Versions.Kubernetes,
+		ControlPlaneEndpoint: controlPlaneEndpoint,
+		APIServer: kubeadmv1beta4.APIServer{
+			ControlPlaneComponent: kubeadmv1beta4.ControlPlaneComponent{
+				ExtraArgs: []kubeadmv1beta4.Arg{
+					{
+						Name:  "enable-admission-plugins",
+						Value: kubeflags.DefaultAdmissionControllers(),
+					},
+					{
+						Name:  "endpoint-reconciler-type",
+						Value: "lease",
+					},
+					{
+						Name:  "kubelet-certificate-authority",
+						Value: "/etc/kubernetes/pki/ca.crt",
+					},
+					{
+						Name:  "profiling",
+						Value: "false",
+					},
+					{
+						Name:  "request-timeout",
+						Value: "1m",
+					},
+					{
+						Name:  "service-node-port-range",
+						Value: cluster.ClusterNetwork.NodePortRange,
+					},
+					{
+						Name:  "tls-cipher-suites",
+						Value: strings.Join(cluster.TLSCipherSuites.APIServer, ","),
+					},
+				},
+				ExtraVolumes: []kubeadmv1beta4.HostPathMount{},
+			},
+			CertSANs: certSANS,
+		},
+		ControllerManager: kubeadmv1beta4.ControlPlaneComponent{
+			ExtraArgs: []kubeadmv1beta4.Arg{
+				{
+					Name:  "flex-volume-plugin-dir",
+					Value: "/var/lib/kubelet/volumeplugins",
+				},
+				{
+					Name:  "profiling",
+					Value: "false",
+				},
+				{
+					Name:  "terminated-pod-gc-threshold",
+					Value: "1000",
```

Comment:
If you wanna make me happy, leave a comment or even make it a constant and explain why 1000 and not 999, or even better, 997.

Reply:
I have zero idea to be honest.